core/sql/generator/GenRelScan.cpp

```diff
@@ -2203,10 +2203,46 @@ short HbaseAccess::codeGen(Generator * generator)
 		       ExpTupleDesc::SQLMX_KEY_FORMAT);
 
   const ValueIdList &retColumnList = retColRefSet_; 
+  // Always get the index name -- it will be the base tablename for
+  // primary access if it is trafodion table.
+  char * tablename = NULL;
+  char * snapshotName = NULL;
+  LatestSnpSupportEnum  latestSnpSupport=  latest_snp_supported;
+  if ((getTableDesc()->getNATable()->isHbaseRowTable()) ||
+      (getTableDesc()->getNATable()->isHbaseCellTable()))
+    {
+      tablename =
+        space->AllocateAndCopyToAlignedSpace(
+                                             GenGetQualifiedName(getTableName().getQualifiedNameObj().getObjectName()), 0);
+      latestSnpSupport = latest_snp_not_trafodion_table;
+    }
+  else
+    {
+      if (getIndexDesc() && getIndexDesc()->getNAFileSet())
+      {
+         tablename = space->AllocateAndCopyToAlignedSpace(GenGetQualifiedName(getIndexDesc()->getNAFileSet()->getFileSetName()), 0);
+         if (getIndexDesc()->isClusteringIndex())
+         {
+            //base table
+            snapshotName = (char*)getTableDesc()->getNATable()->getSnapshotName() ;
+           if (snapshotName == NULL)
+             latestSnpSupport = latest_snp_no_snapshot_available;
+          }
+          else
+            latestSnpSupport = latest_snp_index_table;
```

Comment:
It's outside of this change, but I'm curious: How in general do we guarantee consistency of a table snapshot with index snapshots?